packages/angular_devkit/build_angular/src/dev-server/index.ts

```diff
@@ -397,10 +397,19 @@ export function buildServerConfig(
   const servePath = buildServePath(serverOptions, browserOptions, logger);
   const { styles, scripts } = normalizeOptimization(browserOptions.optimization);
 
-  const config: WebpackDevServer.Configuration & { logLevel: string } = {
+  const headers: {[key: string]: string} = {
+    'Access-Control-Allow-Origin': '*',
+  };
+  if (serverOptions.headers) {
+    for (const [key, value] of Object.entries(serverOptions.headers)) {
+      headers[key] = value;
+    }
+  }
```

Comment:
```suggestion
  const headers: {[key: string]: string} = {
    'Access-Control-Allow-Origin': '*',
    ...serverOptions.headers
  };
```

Reply:
Neat! Applied this directly in the config map.